### PR TITLE
Add Gradle Kotlin DSL example

### PR DESCRIPTION
**Pull request by sugarmanz**
Opening in accordance with https://github.com/jfrog/build-info/pull/441. Note that this was tested with my local snapshot version of the plugin on that PR. Trying to publish the new example will not work unless that PR is released first, or the plugin version is changed to a snapshot version.

### CODE REVIEW

**Comment by clayly** [comment]
I would be very helpful, upvote.

**Comment by clayly** [comment]
But build scripts probably should be updated according to Gradle official codestyle recomendations:
https://docs.gradle.org/current/userguide/kotlin_dsl.htm

For example quote from
https://docs.gradle.org/current/userguide/kotlin_dsl.html#kotdsl:prerequisites

```text
Use of the plugins {} block to declare Gradle plugins significantly improves the editing experience and is highly recommended.
```

File: gradle-examples/gradle-kts-example-publish/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    tasks.named<ArtifactoryTask>("artifactoryPublish") {
+        publications(publishing.publications["ivyJava"])
+    }
+}
+
+configure<org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention> {
+    clientConfig.isIncludeEnvVars = true
+
+    setContextUrl("http://127.0.0.1:8081/artifactory")
+    publish {
```
Comment (clayly) [inline gradle-examples/gradle-kts-example-publish/build.gradle.kts:103]: I don't understand, how it works for you? Look's like Kotlin DSL have no such contructions.

Comment (sugarmanz) [reply gradle-examples/gradle-kts-example-publish/build.gradle.kts:103]: @clayly happy that this is something you're interested in! The Gradle Kotlin DSL example won't work until my other PR to enable such Kotlin DSL support is released:
https://github.com/jfrog/build-info/pull/441

Comment (sugarmanz) [reply gradle-examples/gradle-kts-example-publish/build.gradle.kts:103]: In the meantime, you _can_ still work around this by using the approach load out in this issue:

https://github.com/jfrog/build-info/issues/184

It's very ugly though.

Comment (clayly) [reply gradle-examples/gradle-kts-example-publish/build.gradle.kts:103]: Ok, i got it. I wonder what is happened that there is still no Kotlin DSL support. Keep it going!

**Review by yahavi — APPROVED** [review]

**Pull request merged**